### PR TITLE
Removing dependency on mono on linux.

### PR DESCRIPTION
**Pull request by scabana**
Removes dlls imports to mono dependencies and replaces it an implementation provided by the mono team that does the same thing.

### CODE REVIEW

**Comment by scabana** [comment]
Humm, checked the test in question and it doesn't seem like the submitted code changed anything there. Has it been flaky in the past? I also tried it on my machine, seems to be good.

**Comment by scabana** [comment]
Resolves https://github.com/kblok/puppeteer-sharp/issues/1035.

**Comment by kblok** [comment]
I think we have a real fail on PuppeteerSharp.Tests.PuppeteerTests.FixturesTests.ShouldDumpBrowserProcessStderr on .net 4.7.1 on windows. That failed every time I ran the build

**Comment by scabana** [comment]
I'll take a look.

**Comment by scabana** [comment]
Tests mostly seem to be good. They do timeout on the .net471 run, have this been seen before?

**Comment by kblok** [comment]
@scabana Yeah, Our builds are quite inconsistent in AppVeyor lately. That's why I try to rebuild a few times until I confirm that we might have introduced a bug.

**Comment by scabana** [comment]
Alright! Then, I believe I have fixed the bug (there really was a problem). Handing the next step to you for validation then.

File: lib/PuppeteerSharp/Helpers/Linux/LinuxPermissionsSetter.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            FileAccessPermissions.GroupRead |
+            FileAccessPermissions.GroupExecute |
+            FileAccessPermissions.OtherRead |
+            FileAccessPermissions.OtherExecute;
+
+        public static void SetBrowserPermissionsPermissions(string revisionInfoExecutablePath)
+        {
+            //Extracted this code into a different class so it doesn't get loaded when in .net471.
+
+            var unixFileSystemInfo = UnixFileSystemInfo.GetFileSystemEntry(revisionInfoExecutablePath);
```
[the remark this answers is not in the document]
Comment (scabana) [reply lib/PuppeteerSharp/Helpers/Linux/LinuxPermissionsSetter.cs]: 👍

File: lib/PuppeteerSharp/Helpers/Linux/LinuxPermissionsSetter.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using Mono.Unix;
+
+namespace PuppeteerSharp.Helpers.Linux
+{
+    internal static class LinuxPermissionsSetter
+    {
+        internal static readonly FileAccessPermissions BrowserPermissionsInLinux =
```
[the remark this answers is not in the document]
Comment (scabana) [reply lib/PuppeteerSharp/Helpers/Linux/LinuxPermissionsSetter.cs]: 👍

File: lib/PuppeteerSharp/Helpers/Linux/LinuxPermissionsSetter.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            FileAccessPermissions.UserRead | FileAccessPermissions.UserWrite | FileAccessPermissions.UserExecute |
+            FileAccessPermissions.GroupRead |
+            FileAccessPermissions.GroupExecute |
+            FileAccessPermissions.OtherRead |
+            FileAccessPermissions.OtherExecute;
+
+        public static void SetBrowserPermissionsPermissions(string revisionInfoExecutablePath)
```
[the remark this answers is not in the document]
Comment (scabana) [reply lib/PuppeteerSharp/Helpers/Linux/LinuxPermissionsSetter.cs]: 👍

**Comment by scabana** [comment]
> Good Job!
> Thank you for your contribution :)

Hey, my pleasure! Nice to see an active project! Got an idea of when you're planning next release?

**Comment by kblok** [comment]
Puppeteer released 1.14 today so we'll be working on that during April.
If this is a blocker for you, I can ship 1.13.1.

**Comment by scabana** [comment]
> Puppeteer released 1.14 today so we'll be working on that during April.
> If this is a blocker for you, I can ship 1.13.1.

I think I can live with it. But that task would be far behind when it gets released. If it's not too much work to push a version with those changes, I would greatly appreciate.

**Comment by kblok** [comment]
Thank you for your contribution @scabana. I hope this is the first of many :)

**Comment by kblok** [comment]
Can we review de demo project @scabana ? https://ci.appveyor.com/project/kblok/puppeteer-sharp-0c8w9/builds/23475340/job/vfpp4tm2d222nk7r

**Comment by kblok** [comment]
Issue created @scabana https://github.com/kblok/puppeteer-sharp/issues/1050
Could you give me a hand there?

**Comment by scabana** [comment]
Good morning, sure can do...

**Pull request merged**